qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -28,78 +28,112 @@
-        # post-processed, so that `json` could accept it for further
-        # processing
-        with open("trace_collector.log", "rt") as f:
-            data = f.read()
-            # Removing new lines and tabs around `{`
-            json_string = re.sub("\n\t{\n\t", "{", data)
-            # `resources` field is a dictionary, so adding `{` and`}`
-            # in the next 2 transformations, `instr-lib` is a next field,
-            # so whatever goes before it, belongs to `resources`.
-            json_string = re.sub(
-                "resources     : \n\t", "resources     : {\n\t", json_string
-            )
-            json_string = re.sub(
-                "\n  instr-lib     :", "}\n  instr-lib     :", json_string
-            )
-            # `json`` expects "key":"value" format, some fields in the
-            # data have empty string as value, so need to add `"",`
-            json_string = re.sub(": \n\t", ':"",', json_string)
-            json_string = re.sub(": \n", ':"",', json_string)
-            # Extracted data missing `,' after each key-value pair,
-            # which `json` exppects
-            json_string = re.sub("\n|\n\t", ",", json_string)
-            # Removing tabs
-            json_string = re.sub("\t", "", json_string)
-            # `json` expects each key and value have `"`'s, so adding them to
-            # every word/number/alpha-numeric entry
-            json_string = re.sub(r"\b([\w.-]+)\b", r'"\1"', json_string)
-            # `span kind`` represents one key
-            json_string = re.sub('"span" "kind"', '"span kind"', json_string)
-            # Removing extra `,`
-            json_string = re.sub("{,", "{", json_string)
-            json_string = re.sub(",}", "}", json_string)
-            # Adding `,` between dictionary entries
-            json_string = re.sub("}{", "},{", json_string)
-            # `events` is a list of dictionaries, `json` will accept it in the
-            # form of "events" : [{....}, {.....}, ...]
-            json_string = re.sub(
-                '"events"        : {', '"events"        : [{', json_string
-            )
-            # Closing `events`' list of dictionaries
-            json_string = re.sub('},  "links"', '}],  "links"', json_string)
-            # Last 2 symbols are not needed
-            json_string = json_string[:-2]
-            # Since now `json_string` is a string, which represents dictionaries,
-            # we  put it into one dictionary, so that `json` could read it as one.
-            json_string = '{ "spans" :[' + json_string + "] }"
-            self.spans = json.loads(json_string)["spans"]
+        self.collector_subprocess = subprocess.Popen(
+            ["./otelcol", "--config", "./trace-config.yaml"]
+        )
+
+        time.sleep(5)
+
+        self.filename = "collected_traces.json"
+        self.client_headers = dict(
+            {"traceparent": "00-0af7651916cd43dd8448eb211c12666c-b7ad6b7169242424-01"}
```

Comment:
What's the meaning of this traceparent value? Can you assign to a constant or add a comment?

Reply:
This simulates OTel context being injected on client side: https://www.w3.org/TR/trace-context/#design-overview

Essentially, this is what will be injected to headers/metadata on the client side, and what will be extracted on the server side. The second long value is `trace id`, the third - active `span id`, which should become a parent span.

Since there server side changes are related to extraction of this info through `Propagator`, by simply adding this header during tests, should suffice. I'll later add an example client with OpenTelemetry trace initialization for reference

Reply:
I'll add a comment to tests